### PR TITLE
Switch to uuids

### PR DESCRIPTION
**Pull request by olivier-vadiaval**
Switch from integer id's to uuids
Fix unit tests that depended on id

Regarding some unit tests, assertJSONEqual wouldn't work with responses that had lists. The test was failing even though only the order of items in the list was different. Hence, I had to use a for-loop to check if each item in the response was in the expected items list.

You might need to test on a clean database.

Thank you for fixing the tests for authors and comments and for the initial refactoring, Diego! 🙏

### CODE REVIEW

File: code/socialDistribution/migrations/0005_auto_20211126_1943.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Generated by Django 3.2.8 on 2021-11-27 02:43
+
```
Comment (JawadHossain) [inline code/socialDistribution/migrations/0005_auto_20211126_1943.py:2]: Can you please delete this migration file if not needed.

File: code/socialDistribution/urls.py [hunk 2]
```diff
@@ -11,24 +11,24 @@
...
   path('like-comment/', views.like_comment, name='like-comment'),
-  path('author/<author_id>/posts/<post_id>/comments/', views.post_comment, name='post-comment'),
-  path('delete-post/<int:id>', views.delete_post, name='delete-post'),
-  path('edit-post/<int:id>', views.edit_post, name='edit-post'),
-  path('share-post/<int:id>', views.share_post, name='share-post'),
+  path('author/<author_id>/posts/<str:post_id>/comments/', views.post_comment, name='post-comment'),
+  path('delete-post/<str:id>', views.delete_post, name='delete-post'),
+  path('edit-post/<str:id>', views.edit_post, name='edit-post'),
+  path('share-post/<str:id>', views.share_post, name='share-post'),
   path('copy-link/<int:id>', views.copy_link, name='copy-link')
```
Comment (JawadHossain) [inline code/socialDistribution/urls.py:33]: should be 'copy-link/<str:id>' ?

**Review by JawadHossain — APPROVED** [review]
Looks great, thank you! Just the two comments to address.

**Review by debecerra — APPROVED** [review]
Looks great! Thank you for finishing up with those tests

**Comment by debecerra** [comment]
Please feel free to merge #235 when you merge and deploy this branch

**Pull request merged**